fabric-brewing-v1/src/main/java/net/fabricmc/fabric/api/brewing/PotionIngredient.java

```diff
@@ -0,0 +1,63 @@
+/*
+ * Copyright (c) 2016, 2017, 2018, 2019 FabricMC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.fabricmc.fabric.api.brewing;
+
+import com.google.gson.JsonElement;
+import com.google.gson.JsonObject;
+import com.google.gson.JsonSyntaxException;
+import net.fabricmc.fabric.impl.brewing.PotionIngredientImpl;
+import net.minecraft.item.ItemStack;
+import net.minecraft.potion.Potion;
+import net.minecraft.recipe.Ingredient;
+import net.minecraft.util.Identifier;
+import net.minecraft.util.JsonHelper;
+import net.minecraft.util.PacketByteBuf;
+import net.minecraft.util.registry.Registry;
+
+public interface PotionIngredient {
+	boolean isIngredient();
+	boolean isPotion();
+	Ingredient asIngredient();
+	Potion asPotion();
+	Identifier getPotionType();
+	boolean test(ItemStack stack);
+	void write(PacketByteBuf buf);
```

Comment:
So much to implement. This is depressing to me - it's a lot to implement for custom code. Just like BrewingRecipe. It feels as if it were designed with the JSON files in mind and then the API-able components were just moved to api/. I don't think that's good to encourage.

Reply:
I'll admit - that's because it was. This isn't really the type of thing that needs much of an API layer, but it was my first PR to Fabric and I was a little over-concerned with matching the architecture of existing modules. I should probably go through again and clean this up more, or just drop it since it's a controversial addition anyways.